rust/pattern/constraint/type_constraint.rs

```diff
@@ -200,33 +193,33 @@ impl From<TypeVariable> for PlaysConstraint {
     }
 }
 
-impl Display for PlaysConstraint {
-    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        write!(f, "plays {}", self.role_type)
+impl fmt::Display for PlaysConstraint {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{} {}", Plays, self.role_type)
     }
 }
 
 #[derive(Debug, Clone, Copy, Eq, PartialEq)]
-pub enum IsKey {
+pub enum IsKeyAttribute {
```

Comment:
Actually, I think `IsKeyConstraint` is a better name - we never call `@key` a "key attribute", we call it a key constraint

Reply:
Kind of clashes with the naming scheme. A `FooConstraint` everywhere else is something that applies to a variable.